docs: Extending Cube.js (#1746)
cube-js/cube.js: docs/content/Configuration/Extending-Cubejs.md

---
title: Extending Cube.js
permalink: /configuration/extending-cubejs
category: Configuration
menuOrder: 5
---

> You can install third-party Node.js packages to use inside `cube.js`
> configuration file. Commonly it's used for instrumentation or custom
> authentication logic.

Reviewer: ```suggestion
For some advanced use-cases like instrumentation, or custom
authentication, Cube.js can be extended by installing third-party
Node.js packages and using them in the [`cube.js` configuration file][ref-config].

[ref-config]: /config
```

## Example: sending Cube.js logs to Loggly

> The example below shows how to use Node.js Loggly client to collect and send

Reviewer: ```suggestion
The example below shows how to use the Node.js Loggly client to collect and send
```

logs from Cube.js.

First, you'd need to install the third-party library with [NPM](https://www.npmjs.com/). In our example
we're going to use
[winston-loggly-bulk](https://github.com/loggly/winston-loggly-bulk) to collect

> and send logs to Loggly. You can install it with the following command.

Reviewer: ```suggestion
and send logs to Loggly. You can install it with the following command:
```

```bash
$ npm install --save winston-loggly-bulk
```

[[info | Running Cube.js in Docker]]
| When installing custom Node.js packages for Cube.js running in Docker container, make sure you correctly mount the folder with `node_modules` subfolder.
| ```bash
| $ docker run -d \
|   -v ~/my-cubejs-project:/cube/conf
|   cubejs/cube
| ```
| If you need to use third-party Node.js packages with native extensions, you'd need to build your own Docker image.

> Now we can require and use `winston-loggly-bulk` library inside the `cube.js` file.

Reviewer: ```suggestion
Now we can require and use `winston-loggly-bulk` library inside `cube.js`:
```
